### DISCUSSION TITLE
How to display the error message without returning from main

### DISCUSSION THREAD

**Discussion by chromafish** [zkat/miette#61] [Q&A] [answered]
Sorry if it's a bit silly, but I'm trying to using miette on this toy language that I'm writting, and I'm trying to print the error messages inside of a REPL, but I can't find in the documentation a way of pretty-print the report, beyond lifting the error up to main and exiting the program.

More specifically, I was expecting for this piece of code:
```rust
match interpret(&buf, debug_flag) { // interpret returns miette::Result
    Ok(val) => println!("{:#?}", val),
    Err(err) => println!("{}", err),
}
```

To pretty-print the error with snippets and all, but it just prints the message I defined in the `error` macro. I'm don't know much about Rust so I'm really lost here. I know `miette` uses `thiserror`, which implements `Display`, so I guess this just prints the error from `thiserror`, not miette's stuff? How can I print the full error message?

Hope I could explain my problem clearly, if not, please ask, I would love some help. Miette is pretty cool and being able to have such nice error messages in my toy language feels like a super power :sweat_smile:

**Comment by zkat** [comment, marked as answer]
This should work fine:

```rust
use miette::Context;
let msg = match some_fn().context("Turning this into a Report") {
  Ok(()) => {"ok".to_string()}
  Err(err) => format!("err: {:?}", err),
}
```

**Reply by cylixlee** [reply]
Hi kat, I'm sorry to bother you but there's no `context()` after I `use miette::Context;`. But I managed to find a way to display error message while not exiting the program:

```rust
let report: ErrReport = error.into();
eprintln!("{:?}", report);
```

Is it the right way to print the diagnostic message out? There may be some API changes I guess.

**Reply by zkat** [reply]
That's weird. It should definitely work?

Using `.into()` works fine, too, though.
